packages/interface/src/utils/parseRange.test.ts

```diff
@@ -0,0 +1,61 @@
+/**
+ * Original tests from https://github.com/euank/node-parse-numeric-range
+ */
+
+import { describe, expect, it } from 'vitest';
+import { parseRange } from './parseRange';
+
+function peq(x: string, y: number[]) {
+    expect(parseRange(x)).to.eql(y);
+}
+
+describe('parseRange', function () {
+    it('should parse 1', function () {
+        peq('1', [1]);
+    });
+    it('should parse 1,1', function () {
+        peq('1,1', [1, 1]);
+    });
+    it('should parse 1-5', function () {
+        peq('1-5', [1, 2, 3, 4, 5]);
+    });
+    it('should parse 5-1', function () {
+        peq('5-1', [5, 4, 3, 2, 1]);
+    });
+    it('should parse 1-3,5-6', function () {
+        peq('1-3,5-6', [1, 2, 3, 5, 6]);
+    });
+    it('should parse 10..15', function () {
+        peq('10..15', [10, 11, 12, 13, 14, 15]);
+    });
+    it('should parse 10...15', function () {
+        peq('10...15', [10, 11, 12, 13, 14]);
+    });
+    it('should parse 10..12,13...15,2,8', function () {
+        peq('10..12,13...15,2,8', [10, 11, 12, 13, 14, 2, 8]);
+    });
+    it('should parse ""', function () {
+        peq('', []);
+    });
+    it('should parse -5', function () {
+        peq('-5', [-5]);
+    });
+    it('should parse -5--10', function () {
+        peq('-5--10', [-5, -6, -7, -8, -9, -10]);
```

Comment:
This is very weird to me. I would expect `-5--10` to be `-5, -4, -3, ..., 10`. But I think we should ban negative numbers all together. Since this is used to select which emails to send, it doesn't make sense to send a negative email.

Reply:
Yes, it doesn't make sense to allow negative values for this use case.

And yes, the parseRange function is used to figure out which emails to send, and also to validate the range input, changing the backround color to light red if invalid. To be more specific; if a range is specified, and the length of the return value of parseRange is 0, a css class is added to signify an invalid range.

So going for supporting positive integers and hypens only?

Reply:
It's also possible to validate the value of the react component in the onChange handler with the following regexp. This will ensure only valid ranges can be typed. Seemed to work well when I tested it now. Credit to Claude3.5 for the regexp

```regexp
/^(?:\d+|-?\d+-\d+|-?\d+-|)?(?:,(?:\d+|-?\d+-\d+|-?\d+-|)?)*(?:,)?$/
```

Reply:
I like the idea of highlighting if an invalid range is entered.

Reply:
Validating the input on every change turned out to be much more complex if allowing ranges without a start or end since it is a controlled react input. I.e. then it's necessary to allow a single hyphen, but following the hyphn a range will not be allowed only an integer. Possible to do, but I'm not an regex expert and the AI's I've tried this far needs very careful prompting to get this right, it might be easier to take a deep dive into regex instead. The parseRange function already supported "5-" and I added support for "-5" for ranges in the last commit in this PR.

Reply:
I think it's fine to do in an `onBlur` event by just testing if `parseRange` returns something reasonable.

Reply:
I've merged for now. We can support highlighting range errors in a different PR

Reply:
Yes, that is indirectly how it works now actually. More specific the check is during rendering of Settings-tab.jsx which is triggered by a state update after onBlur.